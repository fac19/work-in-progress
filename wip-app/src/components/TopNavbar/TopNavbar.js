import React from "react";
import { Link } from "react-router-dom";
import { TopNavbarStyle, TopNavbarUlStyle } from "./TopNavbar.style";

const TopNavbar = (props) => {
  return (
    <header>
      <TopNavbarStyle>
        <TopNavbarUlStyle>
          <li>
            <Link to="/new-project">Add new project</Link>
          </li>
          <li>WIP</li>
          <li>
            <Link to="/sign-out">Sign Out</Link>
          </li>
        </TopNavbarUlStyle>
      </TopNavbarStyle>
    </header>
  );
};

export default TopNavbar;
