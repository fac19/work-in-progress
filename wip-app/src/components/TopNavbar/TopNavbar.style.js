import styled from "styled-components";

const TopNavbarStyle = styled.nav`
  height: 4rem;
  border: 1px solid black;
  background-color: white;
  position: fixed;
  width: 100%;
  top: 0;
`;

const TopNavbarUlStyle = styled.ul`
  display: flex;
  flex-direction: row;
  list-style: none;
  justify-content: space-evenly;
  padding: 0;
`;

const TopNavbarLogoStyle = styled.svg`
  height: 2rem;
  .cls-1,
  .cls-3,
  .cls-4,
  .cls-5 {
    fill: none;
    stroke: #231f20;
  }
  .cls-1 {
    stroke-width: 2px;
  }
  .cls-2 {
    fill: #231f20;
  }
  .cls-4 {
    stroke-width: 0.25px;
  }
  .cls-5 {
    stroke-width: 0.75px;
  }
`;

export { TopNavbarStyle, TopNavbarUlStyle, TopNavbarLogoStyle };
