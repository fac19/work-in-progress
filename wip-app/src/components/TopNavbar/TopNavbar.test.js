import React from "react";
import TestRenderer from "react-test-renderer";
import TopNavbar from "./TopNavbar";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";

describe("Snapshot test", () => {
  const history = createMemoryHistory();
  it("renders the correct ui", () => {
    const component = TestRenderer.create(
      <Router history={history}>
        <TopNavbar />
      </Router>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
