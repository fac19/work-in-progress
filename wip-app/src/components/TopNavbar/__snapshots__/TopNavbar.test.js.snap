// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<header>
  <nav
    className="sc-AxjAm eLQCKu"
  >
    <ul
      className="sc-AxirZ dKUKhO"
    >
      <li
        className="sc-AxiKw kzAwcO"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/new-project"
          onClick={[Function]}
        >
          Add project
        </a>
      </li>
      <img
        alt="work in progress logo"
        className="sc-AxhUy eijmgl"
        src="logo.svg"
      />
      <li
        className="sc-AxiKw kzAwcO"
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
        <a
          className="makeStyles-links-1"
          href="/"
          onClick={[Function]}
        >
          Sign Out
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
